lib/hydra/derivatives/image.rb

```diff
@@ -20,37 +20,47 @@ def process_without_timeout
         directives.each do |name, args|
           opts = args.kind_of?(Hash) ? args : {size: args}
           format = opts.fetch(:format, 'png')
-          output_file_name = opts.fetch(:datastream, output_file_id(name))
-          create_resized_image(output_file(output_file_name), opts[:size], format)
+          destination_name = output_filename_for(name, opts)
+          create_resized_image(destination_name, opts[:size], format)
         end
       end
 
+      def output_filename_for(name, opts = {})
+        if opts.has_key? :datastream
+          warn '[Deprecation] datastream option will be removed in a future release.' 
```

Comment:
Please use the Deprecation gem so that it reports where this deprecation is coming from.